engine/storage/src/main/java/org/apache/cloudstack/storage/endpoint/DefaultEndPointSelector.java

```diff
@@ -128,9 +132,24 @@ protected EndPoint findEndPointInScope(Scope scope, String sqlBase, Long poolId)
             }
         }
 
+        List<Long> dedicatedHosts = dedicatedResourceDao.listAllHosts();
```

Comment:
@Spaceman1984 get the dedicated hosts list based on the scope here instead all hosts

Reply:
This call doesn't get all hosts, it gets all dedicated hosts, then builds up a random list of dedicated hosts. To remove some of the entries I will have to iterate through another list of hosts from another resultset which I think would cause more overhead than doing it this way. Or am I misunderstanding your intention?

Reply:
> This call doesn't get all hosts, it gets all dedicated hosts, which is then used to build up a random list of dedicated hosts to move to the back of the result set. To remove some of the entries I will have to iterate through another list of hosts from another resultset which I think would cause more overhead than doing it this way. Or am I misunderstanding your intention?

@Spaceman1984 I mean to say, pick only the dedicated hosts based on the scope. For example, whenever the pool scope is Cluster, pick all the dedicated hosts in that cluster only, not the entire list of dedicated hosts. Got it?

Reply:
Yes, I understand. Making this change is not going to reduce the size of the result set, it will reduce the number of ids to order against. I don't think adding this complexity is worth the effort as this functionality is not available in the DAO, I will have to write a few more functions to make that happen. If you feel that we need this, then I will go ahead.

Reply:
I mean, if a host is not in the order list it gets ignored.

Reply:
@Spaceman1984 New method in dao would be better. Consider a case with 2 clusters, one has 'N' dedicated hosts and the other has None. When you process the pool with second cluster, you have to iterate over the 'N' dedicated hosts (belongs to first cluster) without any use go them. and also, you are passing them to the sql knowingly that these host ids doesn't exist in the result set.